src/Shouldly/Internals/SourceCodeTextGetter.cs

```diff
@@ -84,7 +84,7 @@ string GetCodePart()
             {
                 var codeLines = string.Join("\n", File.ReadAllLines(FileName).Skip(LineNumber).ToArray());
 
-                var indexOf = codeLines.IndexOf(_shouldMethod);
+                var indexOf = codeLines.IndexOf(_shouldMethod!);
```

Comment:
The way to avoid all these dammit operators (`!`) would be to pass the value between methods rather than making it a class field. The state of a mutable class field is also harder to reason about between methods for people as well as compilers, so I would recommend this.